source/core/index.ts

```diff
@@ -1318,7 +1318,7 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			cacheRequest.once('error', reject);
 			cacheRequest.once('request', r => {
 				request = r;
-				resolve(r);
+				r.end();
```

Comment:
Why this change? This seems to be incorrect.

Reply:
I haven't read a lot of got code, I'm still learning how it works internally.
Can you explain me what's the difference between resolving that promise with a request rather than a response?

Reply:
> If yes the correct solution should be putting `resolve(r)` after the `r.end()`.

I read the `_onRequest` code and that solution is wrong, tomorrow I'll look into it.

Reply:
In `cacheable-request` the `request` and `response` events could be emitted before the cache write.

Request event: https://github.com/lukechilds/cacheable-request/blob/master/src/index.js#L156
Response event: https://github.com/lukechilds/cacheable-request/blob/master/src/index.js#L146
Cache write: https://github.com/lukechilds/cacheable-request/blob/master/src/index.js#L131

I can't find any solution to adapt this behavior to got.
I see only 2 options:
The first is to leave it as it is, close (in terms of time) requests might not have the cache.
The second would be working on adding a "cache write" event to `cacheable-request`, to ensure that the got request is resolved only when the cache is successfully written.

I think the best would be the first one, the cache is not a "must", I don't see any problem by having to do a request 2 times (if close in time). I would prefer the speed of the got request, by not waiting a local or __remote__ cache storage.